apps/api/src/Controller/DefaultController.php

```diff
@@ -356,9 +363,23 @@ public function handleMultiGreencheck($data, $ip, $browser, $source, $blind, \St
     {
         $promises = [];
         foreach ($data as $key => $url) {
-
-            $taskdata = ['key' => $url, 'url' => $url, 'ip' => $ip, 'browser' => $browser, 'source' => $source, 'blind' => $blind];
-            $promises[] = $this->producer->sendCommand('greencheck', JSON::encode($taskdata), $needReply = true);
+            $result = $this->getResultFromRedis($url);
+            if ($result) {
+                // set on results
+                $results->$url = JSON::decode($result);
+
+                // push in for later processing
+                $taskdata = ['key' => $url, 'url' => $url, 'ip' => $ip, 'browser' => $browser, 'source' => $source, 'blind' => $blind];
+                $message = new Message(JSON::encode($taskdata));
+                $message->setPriority(MessagePriority::VERY_LOW);
+                $this->producer->sendCommand('greencheck_prio', $message, false);
```

Comment:
Assume that the following url is requested `www.example.com`. It is already in redis so we return the result from redis and then push it to the queue. 1 minute later `www.example.com` is requested again, the url is still in redis, but do we really need to push it to the queue again since that recently happened?

Reply:
that would indeed be a next optimisation. The question being if we still care about the logging of this request. If we still want to count all greenchecks we somehow need to still sent it somewhere.  But we could do that by just putting it on the greencheck_log queue to just log it but don't do another lookup. For me a couple hours or maybe even a day might be a valid cache time. 

But it's a valid remark.

Reply:
> For me a couple hours or maybe even a day might be a valid cache time.

Regarding the cache time. Given that a domain is not likely to change their underlying hosting very often (you typically have your infrastructure at the same place for quite some time) it seems like it would be ok with an even longer cache time.

Reply:
Yes, but we then need a way to reset that cache if the details in the admin system change. If a hostingproviders adds the ip address of this specific site (and others) we dont want it to take weeks or months before that is visible in the result. As we only cache here on website domain and not on the underlying data that makes it green/grey we need to find a way to make this work ok. Resetting all cache results would not be the best way as we would get a cache stampede. 

We might have a flag in redis indicating something changed in admin and if it's less than a day old, every request gets sent through the queue so it gets updated before the next request for an url comes in.

Reply:
I created an issue here: https://github.com/thegreenwebfoundation/Greenwebfoundation-admin/issues/9 where this can be discussed and possibly elaborated on at some later point.